### PR TITLE
fix the bug when alter column that doesn't have a default value to no…

### PR DESCRIPTION
**Pull request by jievince**
…t nullable

## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
Fix https://github.com/vesoft-inc/nebula-ent/issues/1973
Fix https://github.com/vesoft-inc/nebula/issues/4809

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

File: tests/tck/features/schema/Schema.feature [hunk 1]
```diff
@@ -851,6 +837,76 @@ Feature: Insert string vid of vertex and edge
...
+      LOOKUP ON person YIELD properties(VERTEX) AS props;
+      """
+    Then the result should be, in any order, with relax comparison:
+      | props     |
+      | {age: 23} |
+    When executing query:
+      """
+      ALTER TAG person ADD (gender bool NOT NULL);
+      """
+    Then a SemanticError should be raised at runtime: Column `gender' must have a default value if it's not nullable
```
Comment (yixinglu) [inline tests/tck/features/schema/Schema.feature:873]: And drop the used space

File: tests/tck/features/schema/Schema.feature [hunk 2]
```diff
@@ -851,6 +837,76 @@ Feature: Insert string vid of vertex and edge
...
+      LOOKUP ON person YIELD properties(EDGE) AS props;
+      """
+    Then the result should be, in any order, with relax comparison:
+      | props     |
+      | {age: 23} |
+    When executing query:
+      """
+      ALTER EDGE person ADD (gender bool NOT NULL);
+      """
+    Then a SemanticError should be raised at runtime: Column `gender' must have a default value if it's not nullable
```
Comment (yixinglu) [inline tests/tck/features/schema/Schema.feature:908]: ditto

File: tests/tck/features/schema/Schema.feature (hunk 1 above)
Comment (jievince) [reply tests/tck/features/schema/Schema.feature:874]: Good catch

File: src/graph/validator/MaintainValidator.cpp [hunk 3]
```diff
@@ -59,6 +60,12 @@ static Status validateColumns(const std::vector<ColumnSpecification *> &columnSp
     if (!column.nullable_ref().has_value()) {
       column.nullable_ref() = true;
```
Comment (critical27) [inline src/graph/validator/MaintainValidator.cpp:61]: I found this problem recently as well, do you know why? This is quite implicit...

**Review by critical27 — APPROVED** [review]
Good job~

File: src/graph/validator/MaintainValidator.cpp (hunk 3 above)
Comment (jievince) [reply src/graph/validator/MaintainValidator.cpp:61]: Is there any related issue? I don't know which problem you're talking about.

Comment (jievince) [reply src/graph/validator/MaintainValidator.cpp:61]: > I found this problem recently as well, do you know why? This is quite implicit...

Is there any related issue? I don't know which problem you're talking about.

**Review by xtcyclist — DISMISSED** [review]

**Review by xtcyclist — APPROVED** [review]

**Review by critical27 — APPROVED** [review]

**Comment by abby-cyber** [comment]
https://github.com/vesoft-inc/nebula-docs-cn/pull/2503

**Pull request merged**